### PR TITLE
Vagrant: Improve VPC Performance For Windows Server 2022

### LINKED ISSUE #3553: Windows 2022 VagrantPlaybookCheck jobs are taking 11 hours
Ref: https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/OS=Win2022,label=vagrant/buildTimeTrend

This has a significant impact on testing speed, so we should look at whether this can be sped up. While I have not investigated yet if it's from a windows update cycle or Visual Studio installs we should perhaps try to optimise those for the default case.

Unfortunately we don't have time stamps in the logs so this will need to be run manually or just run a job and watch the progress of the log via the jenkins UI.

### PR DESCRIPTION
**Pull request by steelhead31**
Fixes #3553 

Due to poor performance of the Windows Server 2022  vagrant playbook check job, this PR makes a number of improvements to reduce the time taken to build, notably these are:

1) The addition of the "Use Adoptium Box" option, this has only been implemented (currently) for Windows 2022, and when selected will switch the VPC job from using the standard Vagrantfile.Win2022 file, to using the Vagrantfile.Win2022.Adopt file. This 2nd new vagrantfile uses an Adoptium specific vagrant box ( stored on the vagrant server but also imported to all the VPC hosts vagrant boxes ). This vagrant box, is a fully updated ( as of 25th June ) Windows Server 2022 install, so the windows update task in the playbook is much quicker.

2) The 2nd major change, is the selective install of MSVS versions based on the JDK version selected, ( this requires the selection of the fast mode option )

JDK8 will limit the playbook to only installing MSVS_2013
JDK11 will limit the playbook to only installing MSVS_2017
JDK17 will limit the playbook to only installing MSVS_2019
JDK21/JDK22 and JDK head will limit the playbook to only installing MSVS_2022

- [X] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [X] playbook changes run through [VPC](https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptium.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)

### CODE REVIEW

File: ansible/pbTestScripts/vagrantPlaybookCheck.sh [hunk 1]
```diff
@@ -86,6 +89,7 @@ usage()
   --no-halt | -nh                Option to stop the vagrant VMs halting
   --new-vagrant-files | -nv      Use vagrantfiles from the the specified git repository
   --skip-more | -sm              Run playbook faster by excluding things not required by buildJDK
+	--use-adopt | -ua							 Use the local Adoptium vagrantfile instead of the standard (.Adopt extension on Vagrantfile)
```
Comment (karianna) [inline ansible/pbTestScripts/vagrantPlaybookCheck.sh:92]: whitespace?

File: ansible/pbTestScripts/vagrantPlaybookCheck.sh [hunk 2]
```diff
@@ -216,10 +251,11 @@ startVMPlaybook()
 
 	cd $WORKSPACE/adoptopenjdkPBTests/${gitFork}-${newGitBranch}/ansible
 	if [ "$newVagrantFiles" = "true" ]; then
-	  ln -sf vagrant/Vagrantfile.$OS Vagrantfile
+  	ln -sf vagrant/Vagrantfile.$OS Vagrantfile
```
Comment (karianna) [inline ansible/pbTestScripts/vagrantPlaybookCheck.sh:254]: whitespace?  Same below?

File: ansible/vagrant/Vagrantfile.Win2022.Adopt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+# vi: set ft=ruby :
+
+# Runs Powershell as an administator and does the following:
+#  - Gets/executes an Ansible provided script that configures WinRM to allow Ansible to communicate over it.
+#  - Resizes the disk to ~100GB, in line with the 'disksize.size = 100GB' option in the config below
+
+$script = <<SCRIPT
+Start-Process powershell -Verb runAs
+
+
```
Comment (karianna) [inline ansible/vagrant/Vagrantfile.Win2022.Adopt:11]:
```suggestion
```

**Review by karianna — CHANGES_REQUESTED** [review]
Changes look broadly good but suspect tabs vs spaces or some such...

File: ansible/pbTestScripts/vagrantPlaybookCheck.sh (hunk 1 above)
Comment (steelhead31) [reply ansible/pbTestScripts/vagrantPlaybookCheck.sh]: Fixed

File: ansible/pbTestScripts/vagrantPlaybookCheck.sh (hunk 2 above)
Comment (steelhead31) [reply ansible/pbTestScripts/vagrantPlaybookCheck.sh]: Fixed

File: ansible/vagrant/Vagrantfile.Win2022.Adopt (hunk 3 above)
Comment (steelhead31) [reply ansible/vagrant/Vagrantfile.Win2022.Adopt]: Fixed

**Review by karianna — APPROVED** [review]

File: ansible/pbTestScripts/vagrantPlaybookCheck.sh [hunk 4]
```diff
@@ -86,6 +89,7 @@ usage()
   --no-halt | -nh                Option to stop the vagrant VMs halting
   --new-vagrant-files | -nv      Use vagrantfiles from the the specified git repository
   --skip-more | -sm              Run playbook faster by excluding things not required by buildJDK
+	--use-adopt | -ua              Use the local Adoptium vagrantfile instead of the standard (.Adopt extension on Vagrantfile)
```
Comment (karianna) [inline ansible/pbTestScripts/vagrantPlaybookCheck.sh:92]: This one still seems indented wrong

File: ansible/pbTestScripts/vagrantPlaybookCheck.sh [hunk 5]
```diff
@@ -216,10 +251,11 @@ startVMPlaybook()
 
 	cd $WORKSPACE/adoptopenjdkPBTests/${gitFork}-${newGitBranch}/ansible
 	if [ "$newVagrantFiles" = "true" ]; then
-	  ln -sf vagrant/Vagrantfile.$OS Vagrantfile
+    ln -sf vagrant/Vagrantfile.$OS Vagrantfile
```
Comment (karianna) [inline ansible/pbTestScripts/vagrantPlaybookCheck.sh:254]: This one still seems indented wrong

File: ansible/pbTestScripts/vagrantPlaybookCheck.sh [hunk 6]
```diff
@@ -295,10 +331,21 @@ startVMPlaybookWin()
...
+			echo "Use Adoptium Box For Win2022"
+		  ln -sf vagrant/Vagrantfile.$OS.Adopt Vagrantfile
+	  else
+			ln -sf vagrant/Vagrantfile.$OS Vagrantfile
+	  fi
 	else
-	  ln -sf ${scriptPath%/*}/../vagrant/Vagrantfile.$OS Vagrantfile
+		if [[ "$useAdopt" == "true" ]] && [[ "$OS" == "Win2022" ]]; then
+			echo "Use Adoptium Box For Win2022"
+	    ln -sf ${scriptPath%/*}/../vagrant/Vagrantfile.$OS.Adopt Vagrantfile
```
Comment (karianna) [inline ansible/pbTestScripts/vagrantPlaybookCheck.sh:345]: This one still seems indented wrong

**Review by karianna — APPROVED** [review]

**Review by gdams — APPROVED** [review]

**Review by sxa — APPROVED** [review]

**Pull request merged**